doc/source/whatsnew/v0.23.0.txt

```diff
@@ -176,6 +176,57 @@ Please note that the string `index` is not supported with the round trip format,
    new_df
    print(new_df.index.name)
 
+.. _whatsnew_0230.enhancements.index_division_by_zero
+
+Index Division By Zero Fills Correctly
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Division operations on ``Index`` and subclasses will now fill positive / 0 with ``np.inf``, negative / 0 with ``-np.inf``, and 0 / 0 with ``np.nan``.  This matches existing ``Series`` behavior. (:issue:`19322`, :issue:`19347`)
+
+Previous Behavior:
+
+.. code-block:: ipython
+
+    In [6]: index = pd.Index([-1, 0, 1])
+
+    In [7]: index / 0
+    Out[7]: Int64Index([0, 0, 0], dtype='int64')
+
+    In [8]: index = pd.UInt64Index([0, 1])
+
+    In [9]: index / np.array([0, 0], dtype=np.uint64)
+    Out[9]: UInt64Index([0, 0], dtype='uint64')
+
+    In [10]: pd.RangeIndex(1, 5) / 0
+    ---------------------------------------------------------------------------
+    ZeroDivisionError                         Traceback (most recent call last)
+    <ipython-input-10-4c5e91d516f3> in <module>()
+    ----> 1 pd.RangeIndex(1, 5) / 0
+
+    /usr/local/lib/python2.7/site-packages/pandas/core/indexes/range.pyc in _evaluate_numeric_binop(self, other)
+        592                     if step:
+        593                         with np.errstate(all='ignore'):
+    --> 594                             rstep = step(self._step, other)
+        595 
+        596                         # we don't have a representable op
+
+    ZeroDivisionError: integer division or modulo by zero
+
+Current Behavior:
+
+.. ipython:: python
+
+    index = pd.Index([-1, 0, 1])
+    index / 0
+
+    # The result of division by zero should not depend on whether the zero is int or float
```

Comment:
remove this case (or add above as well). the point of the previous/current display is to make it easy to map up what is changing.